### ISSUE TITLE
Fix holes in the native perf map

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#12337]
Trying to perf analyze :
`JIT/Performance/CodeQuality/BenchmarksGame/binarytrees/binarytrees/binarytrees.exe`

When I perf record, I am seeing events occur between functions.  I am trying to figure out what is executed between methog functions.  PreStubs?  Is there a reason we cannot emit these into the perf map.

If I look at the perfmap of a native image:
 `crossgen /Platform_Assemblies_Paths ~/coreoverlay /CreatePerfMap . binarytrees.ni.exe`
```
0000000000021200 2d0 bool [binarytrees] BenchmarksGame.BinaryTrees::Bench(bool)
00000000000214D0 1c4 void [binarytrees] BenchmarksGame.BinaryTrees::Test()
00000000000216A0 50 bool [binarytrees] BenchmarksGame.BinaryTrees::TestBase()
00000000000216F0 3c int32 [binarytrees] BenchmarksGame.BinaryTrees::Main()
0000000000021730 10 instance void [binarytrees] BenchmarksGame.BinaryTrees::.ctor()
0000000000021740 18 instance void [binarytrees] BenchmarksGame.BinaryTrees+TreeNode::.ctor(int32)
0000000000021760 144 valuetype BenchmarksGame.BinaryTrees/TreeNode [binarytrees] BenchmarksGame.BinaryTrees+TreeNode::bottomUpTree(int32,int32)
00000000000218B0 a4 instance void [binarytrees] BenchmarksGame.BinaryTrees+TreeNode::.ctor(valuetype BenchmarksGame.BinaryTrees/TreeNode,valuetype BenchmarksGame.BinaryTrees/TreeNode,int32)
0000000000021960 88 instance int32 [binarytrees] BenchmarksGame.BinaryTrees+TreeNode::itemCheck()
00000000000219F0 10 instance void [binarytrees] BenchmarksGame.BinaryTrees+TreeNode+Next::.ctor()
```

The result is fairly packed except each function is aligned to a 16-byte boundary.

If I look at the map generated by running with the IL. `export COMPlus_PerfMapEnabled=1`
```
0000FFFF1BFC0888 2c int32 [binarytrees] BenchmarksGame.BinaryTrees::Main()
0000FFFF1BFC08D0 40 bool [binarytrees] BenchmarksGame.BinaryTrees::TestBase()
0000FFFF1BFC1360 24c bool [binarytrees] BenchmarksGame.BinaryTrees::Bench(bool)
0000FFFF1BFC17F8 114 valuetype BenchmarksGame.BinaryTrees/TreeNode [binarytrees] BenchmarksGame.BinaryTrees+TreeNode::bottomUpTree(int32,int32)
0000FFFF1BFC1930 70 instance int32 [binarytrees] BenchmarksGame.BinaryTrees+TreeNode::itemCheck()
0000FFFF1BFC32F8 1b8 class [System.Runtime]Microsoft.Win32.SafeHandles.SafeFileHandle [System.Console] dynamicClass::IL_STUB_PInvoke(class [System.Runtime]Microsoft.Win32.SafeHandles.SafeFileHandle)
0000FFFF1BFC34E8 170 int32 [System.Console] dynamicClass::IL_STUB_PInvoke(class [System.Runtime]Microsoft.Win32.SafeHandles.SafeFileHandle,valuetype Interop/Sys/FileStatus&)
0000FFFF1BFC4A50 170 int32 [System.Console] dynamicClass::IL_STUB_PInvoke(class [System.Runtime]Microsoft.Win32.SafeHandles.SafeFileHandle,uint8*,int32)
```
The reported size of each function is smaller.  However, the code sizes in the perfmap seem to match the code size reported by `JitDisasm=*`

There are erratic gaps within the perf map.  The gaps seem to correlate to the difference in size between crossgen and JIT.

@jkotas @janvorli  @brianrob @dotnet/jit-contrib Any Ideas?

**Comment by jkotas** [comment]
> I am trying to figure out what is executed between method functions. PreStubs? 

Yes, they are likely some kind of stubs.

> Is there a reason we cannot emit these into the perf map.

I do not think there is a good reason. Just that nobody has done it.

**Comment by sdmaclea** [comment]
@jkotas Thanks.  I initially was seeing 5% of perf data hitting in these blind spots.  I just collected a trace from an important benchmark whixh had 70% hitting in these blind spots.  I will instrument them.

**Comment by sdmaclea** [comment]
The 70% was a perf bug/misunderstanding, but since I had instrumented the stubs first I have pushed PR #12437

**Comment by sdmaclea** [comment]
Major holes were fixed by #12437
